plugins/modules/identity/bitwarden.py

```diff
@@ -0,0 +1,699 @@
+    item_id:
+        type: str
+        description:
+            - UID of item to target.
+            - Mutually exclusive from item_name.
+        required: false
+        default: null
+    login:
+        type: dict
+        description:
+            - Mapping of keys to values for bw login details such as C(username) and C(password).
+        required: false
+        default: {}
+    notes:
+        type: str
+        description:
+            - A note about the item that is stored as metadata about the item.
+        required: false
+        default: null
+    target:
+        type: str
+        description:
+            - The type of item being added/modified/removed/queried.
+            - For additions/modifications/removal, must be one of C(item) or C(folder).
+            - For queries, the above targets are permitted along with
+                - C(items),
+                - C(folders),
+                - C(organization), and
+                - C(organizations).
+        required: false
+        default: 'item'
+    state:
+        type: str
+        description:
+            - To modify the target, C(absent) or C(present).
+            - To query the state, C(query)
+        required: False
+        default: 'query'
+'''
+
+EXAMPLES = '''
+- name: "Get all organizations"
+  community.general.bitwarden:
+    organization_name: null
+    target: "organizations"
+  delegate_to: "localhost"
+
+- name: "Get all folders"
+  community.general.bitwarden:
+    target: "folders"
+  delegate_to: "localhost"
+
+- name: "Get all items in a folder within an organization"
+  community.general.bitwarden:
+    organization_name: "Test"
+    folder_name: "my_folder"
+    target: "items"
+  delegate_to: "localhost"
+
+- name: "Get an item without an organization in a specific folder"
+  community.general.bitwarden:
+    organization_name: ""
+    folder_name: "my_folder"
+    item_name: "a secret item"
+    target: "items"
+  register: "ret_val"
+  delegate_to: "localhost"
+
+# Note this will not display anything due to no_log being set.
+- name: "Access the password"
+  ansible.builtin.debug:
+    msg: >-
+      {{ ret_val['ansible_module_results']['login']['password'] }}
+
+- name: "Ensure a folder exists"
+  community.general.bitwarden:
+    folder_name: "my_folder"
+    target: "folder"
+    state: "present"
+
+- name: "Ensure an entry exists in the database"
+  community.general.bitwarden:
+    folder_name: "my_folder"
+    item_name: "some item"
+    login:
+      username: "a_user"
+      password: "some password"
+    notes: >-
+      Some information about a user is stored here. foo
+    target: "item"
+    state: "present"
+  delegate_to: "localhost"
+
+- name: "Ensure an entry does not exist in the database"
+  community.general.bitwarden:
+    folder_name: "my_folder"
+    item_name: "some item"
+    target: "item"
+    state: "absent"
+  delegate_to: "localhost"
+'''
+
+RETURN = '''
+---
+# A dictionary with the new/queried state.
```

Comment:
Return values should be explicitly named. Generally returning some arbitrary data returned by another program here on top-level is a very bad idea since this can collide with internal return values used by Ansible, such as `failed`, `changed`, etc.

Reply:
Ah... I put the actual results as the value related to key `ansible_module_results`; is that an issue?

Reply:
Hi @lungj there is more than one aspect to the issue:
* Documentation-wise, this is an empty YAML document, so it provides no information whatsoever to the user. We would like our modules to be better than that - but I am guilty of the same issue elsewhere, so take it with a pinch of salt.
* What @felixfontein refers to is that the documentation should detail each key/value returned by the module, explicitly. See other modules' documentation for examples.
* Last but not least, `ansible_module_results` sounds quite verbose to my ears. Maybe `elements` would be a better reflection of the module's data domain?

Reply:
I actually originally tried to return `results`, but certain versions of Ansible emitted a warning that says the module should, in fact, return `ansible_module_results` and `results` was deprecated. I couldn't figure out what the correct thing to specify was.

The code currently returns something like this:
```python
{
    'changed': True,
    'ansible_module_results': { ...}
}
```
Where the results depend on what is being retrieved/set. I was going by [this document](https://docs.ansible.com/ansible/latest/reference_appendices/common_return_values.html), but it sounds like it doesn't apply in this particular situation. If returning something like `element` is okay in this situation, that does indeed sound much better :)

Reply:
Ok, `results` seems to be reserved. In that case, how about something like `answer`, `result`, `entry`, ...?

Reply:
@lungj see: https://docs.ansible.com/ansible/latest/reference_appendices/common_return_values.html#results